app/src/main/java/com/example/dashcarr/presentation/tabs/settings/RecordingStatisticsFragment.kt

```diff
@@ -76,57 +97,91 @@ class RecordingStatisticsFragment : BaseFragment<FragmentRecordingStatisticsBind
 
         pieChart.invalidate()
 
-        val totalSizeInMB = getTotalAssetsSize(requireContext())
+        var totalElapsedTime: Long = 0
+        val dateFormat = SimpleDateFormat("HH:mm:ss", Locale.getDefault())
+
+        for (i in 0 until jsonArray.length()) {
+            val jsonObject = jsonArray.getJSONObject(i)
+            val elapsedTimeStr = jsonObject.getString("elapsed_time")
+            val elapsedTimeDate = dateFormat.parse(elapsedTimeStr)
+            val elapsedTimeMillis = elapsedTimeDate?.time ?: 0L
+
+            // Convertir le temps en millisecondes vers des secondes
+            val elapsedTimeSeconds = TimeUnit.MILLISECONDS.toSeconds(elapsedTimeMillis)
+
+            totalElapsedTime += elapsedTimeSeconds
+        }
+
+        val averageElapsedTime = if (jsonArray.length() > 0) totalElapsedTime / jsonArray.length() else 0
+
+        val averageElapsedTimeStr = String.format(
+            "%02d:%02d:%02d",
+            TimeUnit.SECONDS.toHours(averageElapsedTime),
+            TimeUnit.SECONDS.toMinutes(averageElapsedTime) % 60,
+            averageElapsedTime % 60
+        )
 
-        binding.inputTotalStorage.text = "%.2f MB".format(totalSizeInMB)
+        binding.inputAvgTime.text = "$averageElapsedTimeStr"
 
-        val numberOfFiles = getNumberOfFilesInAssets(requireContext())
+        val averageSizeFile = averageFileSize(requireContext())
+
+        binding.inputAvgSize.text = "$averageSizeFile"
+
+        val dir = File(requireContext().filesDir, "")
+        val size = getFolderSize(dir)
+        val fileSizeInMB = size.toDouble() / (1024 * 1024)
+
+        binding.inputTotalStorage.text = "%.2f MB".format(fileSizeInMB)
+
+        val files = context?.fileList()
+        val numberOfFiles = files?.size
         binding.inputNumberOfFiles.text = "$numberOfFiles"
     }
 
-    fun getTotalAssetsSize(context: Context): Double {
-        var totalSize = 0.0
-        try {
-            val assetManager = context.assets
-            val files = assetManager.list("")
-            if (files != null) {
-                for (fileName in files) {
-                    var inputStream: InputStream? = null
-                    try {
-                        inputStream = assetManager.open(fileName)
-                        val size = inputStream.available().toDouble()
-                        totalSize += size
-                    } catch (e: Exception) {
-                    } finally {
-                        inputStream?.close()
-                    }
+    fun getFolderSize(folder: File): Long {
```

Comment:
You should overthink your datatype. The function works with long. But it is called only once. There it is instantly casted to Double